### PR TITLE
List Component MVP

### PR DESCRIPTION
**Pull request by jacobrienstra**
So far we have nav and dense styling, and router linking working. Should be enough to start implementing side bars and such. 

Next up will be title subtitle and icons!!!!!!!!!!!!!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "lodash": "^4.17.15",
     "nanoid": "^2.1.11",
     "pinia": "0.0.5",
+    "stylelint-config-prettier": "^8.0.1",
```
Comment (rijkvanzanten) [inline package.json]: What does this do? Why is it cool?

File: .prettierrc [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
-  "htmlWhitespaceSensitivity": "ignore",
-  "printWidth": 100,
-  "singleQuote": true
+    "htmlWhitespaceSensitivity": "ignore",
+    "printWidth": 100,
+    "singleQuote": true,
+    "useTabs": true
```
Comment (rijkvanzanten) [inline .prettierrc]: Wasn't this already the case because it relies on ESLint's config?

File: src/components/v-list/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import VList from './v-list.vue';
+
+export { VList };
```
Comment (rijkvanzanten) [inline src/components/v-list/index.ts]: Should we export the "sub" components here too? Eg VListItem

File: src/components/v-list/v-list-content.vue [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<template functional>
+	<div class="v-list-content">
+		<slot></slot>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+.v-list-item {
+	--v-list-content-padding: 12px 0;
```
Comment (rijkvanzanten) [inline src/components/v-list/v-list-content.vue]: Should we name this `--v-list-padding` instead? That way it matches the actual css property that's being modified

File: src/components/v-list/v-list-item.vue [hunk 5]
```diff
@@ -0,0 +1,112 @@
+<template>
+	<component
+		:is="component"
+		active-class="active"
+		class="v-list-item"
+		:to="to"
+		:class="{ dense, link: isClickable }"
+		v-on="$listeners"
```
Comment (rijkvanzanten) [inline src/components/v-list/v-list-item.vue]: Do you want to pass on every possible listener that's registered?

Comment (rijkvanzanten) [reply src/components/v-list/v-list-item.vue]: (Makes it impossible in the readme to list the available events, as all events of router-link and `li` are possible)

File: src/components/v-list/v-list-item.vue [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+	props: {
+		dense: { type: Boolean, default: false },
+		to: {
+			type: [String, Object] as PropType<string | Location>,
+			default: null
+		}
+	},
+	setup(props, { listeners }) {
+		const component = computed<string>(() => (props.to ? 'router-link' : 'li'));
+		const isClickable = true; //Boolean(props.to || listeners.click !== undefined);
```
Comment (rijkvanzanten) [inline src/components/v-list/v-list-item.vue]:
```suggestion
		const isClickable = computed(() => Boolean(props.to) || listeners.click !== undefined);
```

File: src/components/v-list/v-list-item.vue [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+	--v-list-item-max-width: none;
+	--v-list-item-min-height: 48px;
+	--v-list-item-max-height: auto;
+	--v-list-item-border-radius: 0;
+	--v-list-item-margin-bottom: 0;
+	--v-list-item-color: var(--v-list-color, var(--foreground-color));
+	--v-list-item-color-hover: var(--v-list-color-hover, var(--foreground-color));
+	--v-list-item-color-active: var(--v-list-color-active, var(--foreground-color));
+	--v-list-item-background-color: var(--v-list-background-color, var(--background-color));
+	--v-list-item-background-color-hover: var(
```
Comment (rijkvanzanten) [inline src/components/v-list/v-list-item.vue]: fallback heaven

File: src/components/v-list/v-list-item.vue [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+		var(--hover-background)
+	);
+	--v-list-item-background-color-active: var(
+		--vlist-background-color-active,
+		var(--active-background)
+	);
+
+	position: relative;
+	display: flex;
+	flex: 1 1 100%;
```
Comment (rijkvanzanten) [inline src/components/v-list/v-list-item.vue]: can we make this

```
flex-grow: 1;
flex-shrink: 1;
flex-basis: 100%;
```

to make it a little clearer? I always get confused by the order of this shorthand

File: src/components/v-list/v-list-item.vue [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+	align-items: center;
+	min-width: var(--v-list-item-min-width);
+	max-width: var(--v-list-item-max-width);
+	min-height: var(--v-list-item-min-height);
+	max-height: var(--v-list-item-max-height);
+	margin-bottom: var(--v-list-item-margin-bottom);
+	padding: var(--v-list-item-padding);
+	overflow: hidden;
```
Comment (rijkvanzanten) [inline src/components/v-list/v-list-item.vue]: `text-overflow: ellipses` / `white-space: nowrap` ?

File: src/styles/themes/_default.scss [hunk 10]
```diff
@@ -1,89 +1,91 @@
 /* stylelint-disable custom-property-empty-line-before, declaration-colon-space-after */
 body {
 	/* Global */
-	--brand: 										var(--blue-grey-900);
+	--brand: var(--blue-grey-900);
```
Comment (rijkvanzanten) [inline src/styles/themes/_default.scss]: Aaaaaaaaaaaaaah all the whitespace is gone

File: package.json (hunk 1 above)
Comment (jacobrienstra) [reply package.json]: It's the same as eslint-config-prettier, it prevents stylelint from fighting with prettier

File: .prettierrc (hunk 2 above)
Comment (jacobrienstra) [reply .prettierrc]: You would think, but I kept getting errors so I thought I'd make it explicit

File: src/components/v-list/index.ts (hunk 3 above)
Comment (jacobrienstra) [reply src/components/v-list/index.ts]: Ahh yes good point

File: src/components/v-list/v-list-content.vue (hunk 4 above)
Comment (jacobrienstra) [reply src/components/v-list/v-list-content.vue]: v-list-padding exists though and it's different, that would be the padding for the entire list. This whole file actually needs a rename, should be v-list-item-content.

File: src/components/v-list/v-list-item.vue (hunk 5 above)
Comment (jacobrienstra) [reply src/components/v-list/v-list-item.vue]: I figured a transparent wrapper would be the most useful, so people could use it as a native html element and register whichever listeners they want on there, if they desired. But if you wanna restrict it to click I see no problem with that

File: src/components/v-list/v-list-item.vue (hunk 6 above)
Comment (jacobrienstra) [reply src/components/v-list/v-list-item.vue]: LOL

File: src/components/v-list/v-list-item.vue (hunk 7 above)
Comment (jacobrienstra) [reply src/components/v-list/v-list-item.vue]: truly

File: src/components/v-list/v-list-item.vue (hunk 9 above)
Comment (jacobrienstra) [reply src/components/v-list/v-list-item.vue]: yes, but not yet. those will be in the content and title and subtitle elements

File: src/styles/themes/_default.scss (hunk 10 above)
Comment (jacobrienstra) [reply src/styles/themes/_default.scss]: DAMMIT. IT wasn't when I comIITTEDD

File: src/components/v-list/v-list-item.vue (hunk 8 above)
Comment (jacobrienstra) [reply src/components/v-list/v-list-item.vue]: haha sure

File: src/components/v-list/v-list-item.vue (hunk 5 above)
Comment (rijkvanzanten) [reply src/components/v-list/v-list-item.vue]: Cooooooolio

File: src/components/v-list/v-list.readme.md [hunk 11]
```diff
@@ -36,20 +36,22 @@ You can set the default, active, and hover colors and background colors with css
...
-
 | Slot      | Description      |
 |-----------|------------------|
 | _default_ | List items, etc. |
 
-## CSS Variables
+## Events
+| Event   | Description           | Value        |
+|---------|-----------------------|--------------|
+| `click` | User clicks on button | `MouseEvent` |
```
Comment (jacobrienstra) [inline src/components/v-list/v-list.readme.md]: I'm not sure there are click events on the entire list itself

**Pull request merged**